lib/puppet/agent.rb

```diff
@@ -60,8 +63,17 @@ def run(client_options = {})
               end
             end
           rescue Puppet::LockError
-            Puppet.notice _("Run of %{client_class} already in progress; skipping  (%{lockfile_path} exists)") % { client_class: client_class, lockfile_path: lockfile_path }
-            return
+            wait_for_lock_deadline ||= Time.now.to_i + maxwaitforlock
+
+            if Time.now.to_i > wait_for_lock_deadline
+              Puppet.notice _("Exiting now because the maximum wait for lock time has been exceeded.")
+              nil
+            else
+              Puppet.info _("Another puppet instance is already running")
+              Puppet.info _("Waiting for it to finish. Will try again in %{time} seconds.") % {time: waitforlock}
```

Comment:
With this change `puppet agent -t` command will wait if another puppet is running, is this intended?